src/polygonParts/models/polygonPartsManager.ts

```diff
@@ -301,6 +383,243 @@ export class PolygonPartsManager {
     return { filterQueryMetadata, filteredPolygonPartsQuery };
   }
 
+  private async isValidGeometries(context: {
+    entitiesMetadata: EntitiesMetadata;
+    entityManager: EntityManager;
+    logger: Logger;
+  }): Promise<ValidateError[]> {
+    const {
+      entityManager,
+      logger,
+      entitiesMetadata: {
+        entitiesNames: {
+          validations: { databaseObjectQualifiedName: validationsEntityQualifiedName },
+        },
+      },
+    } = context;
+    logger.debug({ msg: 'Calculating is valid geometries', validationsEntityQualifiedName });
+    try {
+      const rows = await entityManager
+        .createQueryBuilder()
+        .select('table.id', 'id') // alias as "id" so raw object has { id: ... }
+        .from(validationsEntityQualifiedName, 'table')
+        .where('validated = :validated', { validated: false })
+        .andWhere('NOT ST_IsValid(footprint)')
+        .getRawMany<{ id: string }>();
+
+      const result: ValidateError[] = rows.map(({ id }) => ({
+        id,
+        errors: ['ST_IsValid'],
+      }));
+
+      return result;
+    } catch (error) {
+      const errorMessage = `Could not create polygon parts validation table: ${validationsEntityQualifiedName}`;
+      logger.error({ msg: errorMessage, error });
+      throw error;
+    }
+  }
+
+  private async smallGeometriesCount(context: {
+    entitiesMetadata: EntitiesMetadata;
+    entityManager: EntityManager;
+    logger: Logger;
+  }): Promise<ValidationCountSummary> {
+    const {
+      entityManager,
+      logger,
+      entitiesMetadata: {
+        entitiesNames: {
+          validations: { databaseObjectQualifiedName: validationsEntityQualifiedName },
+        },
+      },
+    } = context;
+    logger.info({ msg: 'calculating small area geometries', validationsEntityQualifiedName });
+    try {
+      const [dbResponse] = await entityManager.query<CountQueryResponse[]>(
+        `SELECT * FROM ${this.applicationConfig.validateSmallGeometriesFunction}($1, $2)`,
+        [validationsEntityQualifiedName, this.applicationConfig.validation.areaThresholdSquareMeters]
+      );
+
+      const summary: ValidationCountSummary = {
+        count: dbResponse.count,
+        parts: dbResponse.ids.map((id) => ({
+          id,
+          errors: ['SmallGeometry'],
```

Comment:
please define a closed list for the error keys were going to use (consider raster-shared as enum called ValidationErrorKey)

Reply:
Why would we need this as enum? the worker just takes the strings as is and inserts them to the repost file - there is no further usage to it.
And if you still want an enum it should be in the manager- he is the only one using it

Reply:
i can add this to raster shared now- this service is way behind in the raster-shared versions and i cant upgrade. I can add the enum here and then in the future when will be need someone can add it to raster-shared